src/odemis/gui/cont/tabs.py

```diff
@@ -2622,34 +2745,74 @@ def _perform_switch_position_movement(self, target_button):
         # Only proceed if there is no currently running target_position
         if self._move_future._state == RUNNING:
             return
-        stage = self.tab_data_model.main.stage
-        self._start_pos = stage.position.value
         # Get the required target_position from the pressed button
         self._target_position = next((m for m in self.position_btns.keys() if target_button == self.position_btns[m]),
                                      None)
         if self._target_position is None:
             return
-        # target_position metadata has the end positions for all movements except milling
-        if self._target_position in self.target_position_metadata.keys():
-            if self._current_position is LOADING:
-                box = wx.MessageDialog(self.main_frame, "The sample will be loaded. Please make sure that the sample is properly set and the insertion stick is removed.",
-                                       caption="Loading sample", style=wx.YES_NO | wx.ICON_QUESTION| wx.CENTER)
+        # define the start position 
+        self._start_pos = current_pos = self._stage.position.value
+        self._current_position = getCurrentPositionLabel(current_pos, self._stage)
+
+        if self._role == 'enzel': 
+            # target_position metadata has the end positions for all movements except milling
+            if self._target_position in self.target_position_metadata.keys():
+                if self._current_position is LOADING:
+                    if not self._display_insertion_stick_warning_msg():
+                        return
+                self._end_pos = self.target_position_metadata[self._target_position]
+
+        elif self._role == 'meteor':
+            # determine the end position for the gauge 
+            stage_md = self._stage.getMetadata()
+            if self._current_position == LOADING:
+                # if at loading, set the end poisiton to grid1 center by default 
+                sem_grid1_pos = stage_md[model.MD_SAMPLE_CENTERS][meteor_labels[GRID_1]]
+                if target_button.Label == meteor_labels[SEM_IMAGING]:
+                    self._end_pos = sem_grid1_pos
+                elif target_button.Label == meteor_labels[FM_IMAGING]:
+                    fm_grid1 = transformFromSEMToMeteor(sem_grid1_pos, self._stage.axes)
+                    self._end_pos = fm_grid1
+                    if not self._display_meteor_pos_warning_msg(self._end_pos):
+                        return 
+                elif target_button.Label == meteor_labels[GRID_1]:
+                    self._end_pos = stage_md[model.MD_SAMPLE_CENTERS][meteor_labels[GRID_1]] 
+                elif target_button.Label == meteor_labels[GRID_2]:
+                    self._end_pos = stage_md[model.MD_SAMPLE_CENTERS][meteor_labels[GRID_2]]
```

Comment:
I think now this part is almost identical to going to SEM part, so let's include them together, the only difference is that in Loading-FM move there's no position to convert to Meteor so we take grid 1. 
Btw, we also should display the message whenever we go to FM, not just from loading.